### PR TITLE
#166187815 User is able to Signup

### PR DESCRIPTION
**Pull request by Emmanuelaaron**
#### What does this PR do?

Enables users to signup

#### Description of task completed
* created actions and reducers relevant for signing up a user

#### How should this be manually tested on your local machine
* Do a `git fetch origin ft-signup-166187815`
* Do a `git checkout ft-signup-166187815`
* Run this command `$ npm install`
* Run this command `$ npm start`
* you should be able to view a simple descriptive form that enables you to signup.

finishes [#166187815](https://www.pivotaltracker.com/n/projects/2350340/stories/166187815)

### CODE REVIEW

**Review by alexxsanya — CHANGES_REQUESTED** [review]
@Emmanuelaaron kindly fix this conflict below. I guess rebasing would fix this.
Screenshot 2019-07-17 at 10 00 27

**Review by neelxie — APPROVED** [review]
@Emmanuelaaron I have checked out your feature and have been able to run it on my local and it seems perfect, here is the screenshots.

Screenshot 2019-07-17 at 10 15 23

the response after signing up

Screenshot 2019-07-17 at 10 16 53

the email I have received

Screenshot 2019-07-17 at 10 17 32

the response when I get after the message I get 

Screenshot 2019-07-17 at 10 17 57

**Review by alexxsanya — CHANGES_REQUESTED** [review]
Screenshot 2019-07-18 at 08 28 59
@Emmanuelaaron it seems the reference to the `store` in the `index.jsx` is wrong as it is failing the build on heroku to fail. please make the correct reference then I will be able to proceed with the review. thanks.

**Comment by Emmanuelaaron** [comment]
> Screenshot 2019-07-18 at 08 28 59
> 
> @Emmanuelaaron it seems the reference to the `store` in the `index.jsx` is wrong as it is failing the build on heroku to fail. please make the correct reference then I will be able to proceed with the review. thanks.

Thanks @alexxsanya for taking note of that.
I have fixed it.
You can please continue with the review.
Kind regards

**Review by alexxsanya — CHANGES_REQUESTED** [review]
Screenshot 2019-07-18 at 10 23 32
@Emmanuelaaron I notice you have *postReducer* as a filename in your folder. I suggest you rename the file appropriately in relation to the feature you are working on possibly *signUpReducer.jsx*

**Comment by Emmanuelaaron** [comment]
> Screenshot 2019-07-18 at 10 23 32
> 
> @Emmanuelaaron I notice you have *postReducer* as a filename in your folder. I suggest you rename the file appropriately in relation to the feature you are working on possibly *signUpReducer.jsx*

Noted with thanks.
Already done

**Review by HabibSentongo — CHANGES_REQUESTED** [review]
@Emmanuelaaron kindly improve this styling of the form, the contents go out when errors are displayed.
Screenshot 2019-07-18 at 14 12 39

**Comment by Emmanuelaaron** [comment]
> @Emmanuelaaron kindly improve this styling of the form, the contents go out when errors are displayed.
> Screenshot 2019-07-18 at 14 12 39

I have worked on it.
thanks for taking note of that

**Review by HabibSentongo — APPROVED** [review]
The styling that I told you about is now okay. The functionality also works perfectly well.

**Review by alexxsanya — APPROVED** [review]

**Review by alexxsanya — CHANGES_REQUESTED** [review]
@Emmanuelaaron kindly check out the email validator on the signup form. it seems to accept invalid email format without raising an error as you may see in the screenshot below.
Screenshot 2019-07-18 at 17 36 16
Kindly fix this out so that we can have your PR merged as well.

**Comment by Emmanuelaaron** [comment]
> @Emmanuelaaron kindly check out the email validator on the signup form. it seems to accept invalid email format without raising an error as you may see in the screenshot below.
> Screenshot 2019-07-18 at 17 36 16
> Kindly fix this out so that we can have your PR merged as well.

Hi @alexxsanya , 
Thanks for taking note of that.
I have worked on it.
Screenshot 2019-07-19 at 09 59 33

File: src/auth/signup/CSS/signup.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    color: $primary-color;
+    font-family: $lato-font;
+    font-weight: 100;
+    font-size: 24px;
+  }
+
+  .form {
+    min-height: 500px;
+    width: 544px;
+    border: #cccccc solid 1px;
```
Comment (alexxsanya) [inline src/auth/signup/CSS/signup.scss]: consider defining or using the colors in the `colors.scss` file in the common/variables folder wherever you are using color.

File: src/auth/signup/CSS/signup.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    background-color: #00a4d3;
+    text-align: center;
+    color: white;
+    font-weight: 100;
+    font-family: $lato-font;
+  }
+
+  ::placeholder{
+    color: #000000;
+    font-family: "Source Sans Pro";
```
Comment (alexxsanya) [inline src/auth/signup/CSS/signup.scss:54]: consider using the font variable in the `commons/assets/variables/` folders everywhere in your style files where you are referring to font-family

File: test-reports/clover.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (alexxsanya) [inline test-reports/clover.xml]: @Emmanuelaaron Why do we have this files tracked?

File: src/index.jsx [hunk 4]
```diff
@@ -6,7 +6,9 @@ import store from './Mainstore/Store';
 
 ReactDOM.render(
   <Provider store={store}>
-    <App />
+    <div>
```
Comment (alexxsanya) [inline src/index.jsx]: May I know why you had to change this file by enclose `<App />` under the `<div>` element.

**Review by alexxsanya — CHANGES_REQUESTED** [review]

File: test-reports/lcov-report/index.html [hunk 5]
```diff
@@ -0,0 +1,136 @@
+<!doctype html>
```
Comment (HabibSentongo) [inline test-reports/lcov-report/index.html]: @Emmanuelaaron I see you have unnecessary files that should actually be gitignored, for example, this one and all others in this `test-reports` directory. Kindly consider deleting this from the repository.

**Review by HabibSentongo — CHANGES_REQUESTED** [review]

File: src/auth/signup/redux/actions/signUpAction.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    body: JSON.stringify(postData),
+  })
+    .then(res => res.json())
+    .then((post) => {
+      helper(post);
+      dispatch({
+        type: SIGNUP_USER_SUCCESS,
+        payload: post,
+      }, console.log(post.errors));
```
Comment (HabibSentongo) [inline src/auth/signup/redux/actions/signUpAction.jsx]: @Emmanuelaaron kindly consider removing console logging after you finish what you needed it for during development.

**Review by alexxsanya — APPROVED** [review]

File: src/auth/signup/CSS/signup.scss (hunk 1 above)
Comment (Emmanuelaaron) [reply src/auth/signup/CSS/signup.scss]: I have worked on it.

**Review by jennizalwango — COMMENTED** [review]
Screenshot 2019-07-19 at 19 08 39

@Emmanuelaaron knidly note to change this . Reduce on the space between the input fileds and also reduce on the height of the form

File: src/auth/signup/CSS/signup.scss (hunk 2 above)
Comment (Emmanuelaaron) [reply src/auth/signup/CSS/signup.scss:54]: Resolved

File: test-reports/clover.xml (hunk 3 above)
Comment (Emmanuelaaron) [reply test-reports/clover.xml]: I have removed it

File: src/index.jsx (hunk 4 above)
Comment (Emmanuelaaron) [reply src/index.jsx]: This inside the div tag

File: test-reports/lcov-report/index.html (hunk 5 above)
Comment (Emmanuelaaron) [reply test-reports/lcov-report/index.html]: I have resolved it @HabibSentongo.
Thank you for taking note of that

**Comment by Emmanuelaaron** [comment]
> Screenshot 2019-07-19 at 19 08 39
> 
> @Emmanuelaaron knidly note to change this . Reduce on the space between the input fileds and also reduce on the height of the form

Screenshot 2019-07-19 at 19 23 44
Thank you @jennizalwango, I have worked on it.

**Review by jennizalwango — APPROVED** [review]

**Review by HabibSentongo — APPROVED** [review]
All feedback has been implemented and functionality is working fine. We are good to go.

File: src/auth/signup/components/signUp.jsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import signUpUser from '../redux/actions/signUpAction';
+import validation from '../validation';
+import '../CSS/signup.scss';
+
+const Loader = require('react-loader');
```
Comment (AGMETEOR) [inline src/auth/signup/components/signUp.jsx]: @Emmanuelaaron what happens if we instead import like so `import Loader from 'react-loader` ?

File: src/auth/signup/components/signUp.jsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+export const SignUpUserForm = ({ signUpUser: signUp, isLoading }) => {
+  const [fields, setFields] = useState({ username: '', email: '', password: '' });
+  const [touched, setTouched] = useState({ username: false, email: false, password: false });
+
+  const onSubmit = (e) => {
+    const post = {
```
Comment (AGMETEOR) [inline src/auth/signup/components/signUp.jsx]: Use a more meaningful name for this object, something like `userData` would suffice?

File: src/auth/signup/components/signUp.jsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  );
+};
+
+SignUpUserForm.propTypes = {
+  signUpUser: PropTypes.func.isRequired,
+  isLoading: PropTypes.bool.isRequired,
+};
+
+const mapStateToProps = state => ({
+  // signUpUser: state.user,
```
Comment (AGMETEOR) [inline src/auth/signup/components/signUp.jsx]: Remove commented out code.

File: src/index.jsx (hunk 4 above)
Comment (AGMETEOR) [reply src/index.jsx]: @Emmanuelaaron I also don't understand why you had to make that change. Is it necessary? If not, remove it.

**Review by AGMETEOR — CHANGES_REQUESTED** [review]

**Comment by AGMETEOR** [comment]
@Emmanuelaaron all components and containers file names start with capital letters. Look at the code on LandingPage branch for reference on this.

File: src/auth/signup/components/signUp.jsx (hunk 8 above)
Comment (Emmanuelaaron) [reply src/auth/signup/components/signUp.jsx]: okay,
I've done that and I've realized it's better that way.

File: src/auth/signup/components/signUp.jsx (hunk 9 above)
Comment (Emmanuelaaron) [reply src/auth/signup/components/signUp.jsx]: oh yeah.
Thanks for taking note of that

**Comment by Emmanuelaaron** [comment]
> @Emmanuelaaron all components and containers file names start with capital letters. Look at the code on LandingPage branch for reference on this.

I've worked on it

File: src/auth/signup/components/signUp.jsx (hunk 7 above)
Comment (Emmanuelaaron) [reply src/auth/signup/components/signUp.jsx]: According to the `react-loader` [resource](https://www.npmjs.com/package/react-loader), you can only declare a variable of `Loader` to a `require` class with react `react-loader as an attribute

**Comment by alexxsanya** [comment]
Screenshot 2019-07-22 at 13 58 09
@HabibSentongo as per the coverage report, it seems like you have not written a single test for the code you have added. please consider writing tests.

**Review by AGMETEOR — APPROVED** [review]

**Pull request merged**